packages/broccoli-eyeglass/src/index.ts

```diff
@@ -175,6 +151,26 @@ class EyeglassCompiler extends BroccoliSassCompiler {
     if (this.configureEyeglass) {
       this.configureEyeglass(eyeglass, options.eyeglass.engines.sass, details);
     }
+
+    // set up asset dependency tracking
+    eyeglass.assets.resolver((filepath, fullUri, realResolve, cb)  => {
+      this.events.emit("dependency", filepath).then(() => {
+        realResolve(filepath, fullUri, cb);
+      }, cb);
```

Comment:
if `realResolve` throws the `cb` errback wont be notified, is that intentional?

I would expect:

```js
then(() => realResolve(filePath, fullUri, cb)).catch(cb);
```


An idea for the future, improving the eyeglass apis to natively support promises could result in:

```js
eyeglass.assets.resolver(async (filepath, fullUri) => {
  await this.events.emit('dependency');
  return realResolve(filepath, fullUri);
});

Reply:
The contract for resolve is to report all errors via the callback, so it shouldn't throw. Is this just out of extreme caution for code that doesn't comply with the contract?